docs/src/lens.md

```diff
@@ -0,0 +1,276 @@
+# Lenses
+
+Lets say we're building a todo list application, and we are designing the widget
+that will represent a single todo item. Or data model looks like this:
+
+```rust
+/// A single todo item.
+#[derive(Clone, Data)]
+struct TodoItem {
+    title: String,
+    completed: bool,
+    urgent: bool,
+}
+```
+
+And we would like our widget to display the title of the item, and then below
+that to display two checkmarks that toggle the 'completed' and 'urgent' bools.
+`Checkbox` is a struct that implements `Widget<bool>`. How do we use it with
+`TodoItem`? By using a `Lens`.
+
+## Conceptual
+
+You can think of a lens as a way of "focusing in" on one part of the data. You
+have a `TodoItem`, but you *want* a `bool`.
+
+`Lens` is a trait for types that perform this "focusing in" (aka *lensing*).
+A simplified version of the `Lens` trait might look like this:
+
+```rust
+trait SimpleLens<T, U> {
+    fn focus(&self, data: &T) -> U;
+}
+```
+
+That is, this type takes an instance of `T`, and returns an instance of `U`.
```

Comment:
Maybe use `In` and `Out` instead of `T` and `U` to keep it the same with the second example?

Reply:
good idea, thanks!